### ISSUE TITLE
`@typescript-eslint/strict-boolean-expressions` increases code x5 with little improvement

### ISSUE THREAD

**Issue by fregante** [xojs/eslint-config-xo#108]
This is just one of the many situations reported by the rule:

<img width="597" height="69" alt="Image" src="https://github.com/user-attachments/assets/0d59553e-533e-4b57-85b7-091c3f3a1f23" />

I think the rule has some merit but in practice it has too many false positives for any situation where "falsy" is completely intentional.

**Comment by sindresorhus** [comment]
The only change we could do, other than turning off the rule, is change `allowString`/`allowNumber`, and I'd rather keep this strict. The `allowString`/`allowNumber` cases catch real bugs (`''`, `0`, `NaN` are common valid values that silently read as falsy), and the fix is a one-liner like `str !== ''` or `array.length > 0` that's clearer about intent anyway.

Your specific example is a different thing: `isRepoCommitList() && $optional(…)` is typed `false | Element | undefined`, which spans both the boolean and object groups, so no `allow*` option can cover it. The clean fix is to make the intent a ternary:

```ts
const feedLink = isRepoCommitList() ? $optional('link[type="application/atom+xml"]') : undefined;
if (!feedLink) { … }
```

That gives `Element | undefined` (allowed) and reads better than storing `false` to mean "n/a". I'd keep the rule and rewrite these cases rather than loosen it.

**Comment by fregante** [comment]
Even with allowString/allowNumber it found 114 reports in Refined GitHub, reporting conditions like:

```js
async function confirmAndReload(_menuInfo: unknown, tab: chrome.tabs.Tab): Promise<void> {
	if (tab.id && tab.url && await isContentScriptRegistered(tab.url)) {
		await startFeatureIdentification(new URL(tab.url).origin);
		await chrome.tabs.reload(tab.id);
	} else {
		await alert('Refined GitHub is not running on this page');
	}
}
```

which to me confirms that the rule just wants to turn acceptable code (which is already lengthy due as a result of TS null checks) into overly verbose code
